lib/publiccloud/ecr.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::ecr;
```

Comment:
we have a [**test**](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/upload_image.pm) which is responsible for uploading images . Depending on provider set via variables it initialize appropriate provider class and do upload . I think it should be done same way here .